lib/sidekiq/client.rb

```diff
@@ -70,7 +70,7 @@ def initialize(redis_pool = nil)
     #   push('queue' => 'my_queue', 'class' => MyWorker, 'args' => ['foo', 1, :bat => 'bar'])
     #
     def push(item)
-      normed = normalize_item(item)
+      normed = item.key?("normalized") ? item : normalize_item(item)
```

Comment:
Can you explain the reason for adding this `normalized` key? I'm guessing that it's related to the backwards compatibility that you mention in the description, but I still don't get why it's actually needed.

Reply:
There are many places throughout the Sidekiq codebase that use `Client#push` outside of the context of a Job instance. Those places need to still work, thus they need to follow the original codepath. But, inside the context of a Job instance, I want `Client#push` to go down a new code path (not calling `normalize_item` being a chief difference). In order to allow `Client#push` to determine which code path to go down, I created this mechanism. So, whenever the `item` payload passed to `Client#push` has already called `normalize_item` _before_ calling `Client#push`, that can be known.

Of course, I could have leaned on the particulars of how `normalize_item` mutates the `item` hash to introspect inside of `Client#push`. I didn't want to create such a hard (and hard to notice) dependency between those two parts of the codebase.

Reply:
Makes sense, thanks for explaining 👍